File: sdks/kotlin/lib/src/main/kotlin/OperationalWebhook.kt
package com.hooksniff.kotlin

class OperationalWebhook(private val client: HookSniffHttpClient) {
    fun list(): List<OperationalWebhookEndpointOut> = client.executeRequest("GET", "/api/v1/operational-webhooks")
    fun create(body: OperationalWebhookEndpointIn): OperationalWebhookEndpointOut = client.executeRequest("POST", "/api/v1/operational-webhooks", body)
    fun get(id: String): OperationalWebhookEndpointOut = client.executeRequest("GET", "/api/v1/operational-webhooks/$id")
    fun update(id: String, body: OperationalWebhookEndpointIn): OperationalWebhookEndpointOut = client.executeRequest("PUT", "/api/v1/operational-webhooks/$id", body)
    fun delete(id: String) { client.executeRequest<Any>("DELETE", "/api/v1/operational-webhooks/$id") }
    fun listDeliveries(id: String): List<OperationalWebhookDeliveryOut> = client.executeRequest("GET", "/api/v1/operational-webhooks/$id/deliveries")
}

File: sdks/go/operational_webhook.go
package hooksniff

import (
	"context"
	"github.com/servetarslan02/hooksniff-go/internal"
	"github.com/servetarslan02/hooksniff-go/models"
)

type OperationalWebhook struct{ client *internal.HookSniffHttpClient }

func newOperationalWebhook(client *internal.HookSniffHttpClient) *OperationalWebhook {
	return &OperationalWebhook{client: client}
}

func (o *OperationalWebhook) List(ctx context.Context) ([]models.OperationalWebhookEndpointOut, error) {
	var r []models.OperationalWebhookEndpointOut
	err := o.client.Get(ctx, "/api/v1/operational-webhooks", nil, &r)
	return r, err
}

func (o *OperationalWebhook) Create(ctx context.Context, body *models.OperationalWebhookEndpointIn) (*models.OperationalWebhookEndpointOut, error) {
	var r models.OperationalWebhookEndpointOut
	err := o.client.Post(ctx, "/api/v1/operational-webhooks", nil, body, &r)
	return &r, err
}

func (o *OperationalWebhook) Get(ctx context.Context, id string) (*models.OperationalWebhookEndpointOut, error) {
	var r models.OperationalWebhookEndpointOut
	err := o.client.Get(ctx, "/api/v1/operational-webhooks/"+id, nil, &r)
	return &r, err
}

func (o *OperationalWebhook) Update(ctx context.Context, id string, body *models.OperationalWebhookEndpointIn) (*models.OperationalWebhookEndpointOut, error) {
	var r models.OperationalWebhookEndpointOut
	err := o.client.Put(ctx, "/api/v1/operational-webhooks/"+id, nil, body, &r)
	return &r, err
}

func (o *OperationalWebhook) Delete(ctx context.Context, id string) error {
	return o.client.Delete(ctx, "/api/v1/operational-webhooks/"+id, nil)
}

func (o *OperationalWebhook) ListDeliveries(ctx context.Context, id string) ([]models.OperationalWebhookDeliveryOut, error) {
	var r []models.OperationalWebhookDeliveryOut
	err := o.client.Get(ctx, "/api/v1/operational-webhooks/"+id+"/deliveries", nil, &r)
	return r, err
}
